javascripts/discourse/api-initializers/table-editor.js

```diff
@@ -59,7 +59,11 @@ export default apiInitializer("0.11.1", (api) => {
       if (table.parentNode.contains(expandBtn)) {
         expandBtn.parentNode.insertBefore(popupBtn, expandBtn);
       } else {
-        table.parentNode.insertBefore(popupBtn, table);
+        const buttonWrapper = document.createElement("div");
+        buttonWrapper.classList.add("fullscreen-table-wrapper-buttons");
+        buttonWrapper.append(popupBtn);
+        popupBtn.addEventListener("click", generateModal, false);
+        table.parentNode.insertBefore(buttonWrapper, table);
```

Comment:
Can you help me to understand why we need to duplicate the logic in core in this plugin here? I think this implementation pattern is brittle because if there are changes in core, this plugin will have to change as well. Perhaps it is worth thinking about how this plugin can properly extend core through an interface.

Reply:
@tgxworld 
This is for the case of a table in a post that doesn't have the <kbd>Expand Button</kbd> (this happens for tables that are not very large). The component doesn't necessarily need the button wrapper as it has in core, especially when its the case of only the <kbd>Edit Table</kbd> button being present, but I'm just copying the logic in core, so I can inherit the same styles and such.. I could omit this and keep it as before and it should work fine.

I'm open to changing things though, please let me know your thoughts.

Reply:
> but I'm just copying the logic in core,

If the logic in core changes, is there a chance this could break the plugin feature/UX wise. Duplicating core's logic in a plugin is bound to be brittle so if a plugin relies on a certain behaviour of core, we should define a plugin api such that the plugin can rely on the logic in core.

Reply:
Thanks @tgxworld, that's a good point. I'm now working on adding a new pluginAPI in the corresponding PR in core (https://github.com/discourse/discourse/pull/18810)

Reply:
I'm going to resolve this for now so we can get rid of the post-height issues and include these changes with the [PR fix](https://github.com/discourse/discourse/pull/18810) in core. Adding it to my list when I have more time to revisit the `pluginAPI` addition to improve the logic in core and extension in this component, but for now I'd like to wrap these PRs up.